### PR TITLE
.github: update codeowners

### LINKED ISSUE #25077: Update CODEOWNERS file
Our [CODEOWNERS](https://github.com/pingcap/tidb/blob/master/.github/CODEOWNERS) file is out-of-date for quite a while. Also the team relationship is lack of maintenance. In order to

1. Keep CODEOWNERS up-to-date and thus review auto-assignee works smoothly
2. Decouple CODEOWNERS with team maintenance which is out of regular workflow.

I'd propose we update CODEOWNERS file and break down owners into individuals.

* [x] #25814

### PR DESCRIPTION
**Pull request by tisonkun**
Signed-off-by: tison <wander4096@gmail.com>

### What problem does this PR solve?

Issue Number: close #25077

Problem Summary:

### What is changed and how it works?

What's Changed:

Update CODEOWNERS file.

How it Works:

1. Keep CODEOWNERS up-to-date and thus review auto-assignee works smoothly
2. Decouple CODEOWNERS with team maintenance which is out of regular workflow.

### Release note 

- No release note

### CODE REVIEW

**Comment by morgo** [comment]
Can I (or sql-infra) be a codeowner for `/sessionctx/variable`?

**Comment by tisonkun** [comment]
> Can I (or sql-infra) be a codeowner for `/sessionctx/variable`?

Updated. Actually I'd like to hear from members in the file to reach a consensus on current state. You can use suggestion feature directly modify the changes though.

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
+/server @coocood @morgo
+/session @cfzjywxk @tiancaiamao
+/sessionctx @qw4990
+/sessionctx/variable @morgo
+/statistics @winoros @eurekaka
+/store @disksing @AndreMouche @MyonKeminta
+/structure @zimulala
+/table @wjhuang2016
+/tablecodec @tangenta
+/telemetry @breeswish @lzmhhh123
```
Comment (breeswish) [inline .github/CODEOWNERS:29]:
```suggestion
/telemetry @breeswish @lzmhhh123 @kaixu120811
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
+/session @cfzjywxk @tiancaiamao
+/sessionctx @qw4990
+/sessionctx/variable @morgo
+/statistics @winoros @eurekaka
+/store @disksing @AndreMouche @MyonKeminta
+/structure @zimulala
+/table @wjhuang2016
+/tablecodec @tangenta
+/telemetry @breeswish @lzmhhh123 @kaixu120811
+/tests @ichn-hu
```
Comment (ichn-hu) [inline .github/CODEOWNERS:30]: why am I the code owner of `/tests`?

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
-/owner @pingcap/co-ddl
+/bindinfo @rebelice
+/cmd @eurekaka @rebelice
+/config @tiancaiamao @qw4990
+/ddl @zimulala @tiancaiamao
+/distsql @crazycs520 @qw4990
+/docs @zz-jason @bb7133
+/domain @lysu @xhebox
+/errno @morgo @tangenta
+/executor @lysu @XuHuaiyu @wshwsh12
```
Comment (ichn-hu) [inline .github/CODEOWNERS:9]:
```suggestion
/executor @lysu @XuHuaiyu @wshwsh12 @ichn-hu 
```

**Comment by tisonkun** [comment]
closed for now as before we find explicit auto assign problem on #25077

**Comment by tisonkun** [comment]
/hold
/uncc @zz-jason @bb7133 @XuHuaiyu @winoros
keep open for comment, but do not merge

File: .github/CODEOWNERS (hunk 2 above)
Comment (tisonkun) [reply .github/CODEOWNERS:30]: It is based on 

```
$ find -EL tests -type f -regex '.*\.(md|go)' | xargs -n 1 git blame --porcelain | grep  "^author " | sort | uniq -c | sort -nr | head -10
   3 author Ping Yu
   2 author 虎
   2 author jianzhiyao
   2 author Tjianke
   2 author Jack Yu
   1 author Zhou Kunqin
   1 author Yuanjia Zhang
   1 author Feng Liyuan
```

but yes. We might think of the real problem on auto assign. Possibly make use of https://book.prow.tidb.io/#/plugins/blunderbuss cc @hi-rustin @Mini256

File: .github/CODEOWNERS (hunk 3 above)
Comment (tisonkun) [reply .github/CODEOWNERS:9]: updated.

**Comment by tisonkun** [comment]
After an offline discussion there is a consensus that CODEOWNERS does not well fit our codebase and actual owners mapping. Thus, I plan to create another PR to remove the CODEOWNERS.

**Pull request closed without merging**